src/control_devices/CCPACSLeadingEdgeDevice.cpp

```diff
@@ -0,0 +1,335 @@
+/*
+* Copyright (C) 2025 German Aerospace Center (DLR/SC)
+*
+* Created: 2025 Ole Albers <ole.albers@dlr.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "CCPACSLeadingEdgeDevice.h"
+#include "generated/CPACSControlSurfacePath.h"
+#include "generated/CPACSControlSurfaceStep.h"
+#include "CTiglControlSurfaceTransformation.h"
+#include "CTiglFusePlane.h"
+
+#include "tiglcommonfunctions.h"
+#include "tiglmathfunctions.h"
+#include "tigletaxsifunctions.h"
+
+// TODO: Replace by TiGL function
+#include <BRepBuilderAPI_Transform.hxx>
+
+namespace tigl
+{
+
+CCPACSLeadingEdgeDevice::CCPACSLeadingEdgeDevice(CCPACSLeadingEdgeDevices* parent, CTiglUIDManager* uidMgr)
+    : generated::CPACSLeadingEdgeDevice(parent, uidMgr)
+    , CTiglAbstractGeometricComponent()
+    , m_hingePoints(*this, &CCPACSLeadingEdgeDevice::ComputeHingePoints)
+    , m_cutoutShape(*this, &CCPACSLeadingEdgeDevice::ComputeCutoutShape)
+    , m_flapShape(*this, &CCPACSLeadingEdgeDevice::ComputeFlapShape)
+    , m_type(LEADING_EDGE_DEVICE)
+{
+}
+
+void CCPACSLeadingEdgeDevice::ReadCPACS(const TixiDocumentHandle& tixiHandle, const std::string& xpath)
+{
+    CPACSLeadingEdgeDevice::ReadCPACS(tixiHandle, xpath);
+
+    m_currentControlParam = Clamp(0., GetMinControlParameter(), GetMaxControlParameter());
+}
+
+// get short name for loft
+std::string CCPACSLeadingEdgeDevice::GetShortName() const
```

Comment:
This function seems to be implemented to work for spoilers, teds and leds, but really it can only ever be called for LEDs. 

Suggestion for more DRYness:

CCPACSControlSurfaces could provide a method that generates a short name from a controldevice uid. Then this function, and the corresponding function in CCPACSTrailingEdgeDevice can call the helper method.